### PR TITLE
#2145. Update assertions and descriptions in the current Local variable declaration tests

### PR DESCRIPTION
**Pull request by sgrekhov**
There will be two more PRs. Ther next one will rename the tests according to the updated assertions and delete duplicates. Another one will add missing tests

### CODE REVIEW

**Comment by eernstg** [comment]
OK, sounds good!

File: Language/Statements/Local_Variable_Declaration/syntax_t01.dart [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks several assignments of variables declared as "T id;"
-/// that are correct and do not produce static warnings.
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t01.dart:7]: I think it's good for the overall readability if we preserve the spaces:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t02.dart [hunk 2]
```diff
@@ -2,16 +2,13 @@
...
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks several assignments of variables declared as "T id = e;"
-/// that are correct and do not produce static warnings.
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t02.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t03.dart [hunk 3]
```diff
@@ -2,15 +2,14 @@
...
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks that a variable declaration statement T id;  must end
-/// with a semicolon.
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t03.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’

```

File: Language/Statements/Local_Variable_Declaration/syntax_t03.dart [hunk 4]
```diff
@@ -2,15 +2,14 @@
...
+///
+/// @description Checks that a variable declaration statement `T id;`  have a
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t03.dart:9]:
```suggestion
/// @description Checks that a variable declaration statement `T id;` has a
```

File: Language/Statements/Local_Variable_Declaration/syntax_t05.dart [hunk 5]
```diff
@@ -2,12 +2,15 @@
...
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks that the variable 'int id;' declared in a block is not
-/// available in a scope that encloses this block.
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t05.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t06.dart [hunk 6]
```diff
@@ -2,14 +2,16 @@
...
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks that the variable declared 'int id;' in a block is not
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t06.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t07.dart [hunk 7]
```diff
@@ -2,15 +2,16 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t07.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t08.dart [hunk 8]
```diff
@@ -2,14 +2,16 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t08.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t12.dart [hunk 9]
```diff
@@ -2,18 +2,16 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t12.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

File: Language/Statements/Local_Variable_Declaration/syntax_t13.dart [hunk 10]
```diff
@@ -2,15 +2,14 @@
...
 
-/// @assertion A variable declaration statement declares a new local variable.
-///   localVariableDeclaration:
-///     initializedVariableDeclaration ’;’
-///   ;
-/// @description Checks that incorrect syntax of type parameters causes compile
-/// error.
+/// @assertion A variable declaration statement, also known as a local variable
+/// declaration, has the following form:
+/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩⟨initializedVariableDeclaration⟩‘;’
```
Comment (eernstg) [inline Language/Statements/Local_Variable_Declaration/syntax_t13.dart:7]:
```suggestion
/// ⟨localVariableDeclaration⟩ ::= ⟨metadata⟩ ⟨initializedVariableDeclaration⟩ ‘;’
```

**Review by eernstg — COMMENTED** [review]
Looks good, I'm just suggesting a few whitespace fixes (to ensure that the grammar rules look similar to how they look elsewhere).

**Comment by sgrekhov** [comment]
Updated

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**